src/Nethermind/Nethermind.Pipeline/Pipeline.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using Nethermind.Pipeline.Publishers;
+using Nethermind.PubSub;
+
+namespace Nethermind.Pipeline
+{
+    public class Pipeline<T> : IPipeline<T>
+    {
+        public Stack<IPipelineElement<T>> Elements { get; }
+        public ISource<T> Source { get; }
+
+        public Pipeline(ISource<T> source)
+        {
+            Source = source ?? throw new ArgumentNullException(nameof(source));
+        }
+
+        public void AddElement(IPipelineElement<T> element)
+        {
+            var lastElement = Elements.Peek();
+            if (lastElement == null)
+            {
+                Source.Emit = element.SubscribeToData;
```

Comment:
I think this is a replacement fo events? is that some way suggested in reactive extensions or something?

Reply:
Yeah at first Emit was an EventHandler but my idea was that since we only get one receiver it would look simpler without += and -= there